### ISSUE TITLE
The Node-API seems to have a maximum of 6 arguments 

### ISSUE THREAD

**Issue by jean343** [oven-sh/bun#6841]
### What version of Bun is running?

1.0.7+b0393fba6200d8573f3433fb0af258a0e33ac157

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

Using the base example of https://www.npmjs.com/package/duckdb, increase the number of arguments until it fails.

This example works perfectly fine
```js
var duckdb = require("duckdb");
var db = new duckdb.Database(":memory:");
db.all("SELECT ?,?,?,?", 1, 2, 3, 4, function (err, res) {
  if (err) {
    throw err;
  }
  console.log(res);
});
```

While this one never completes, the callback isn't called. Notice the extra argument.
```js
var duckdb = require("duckdb");
var db = new duckdb.Database(":memory:");
db.all("SELECT ?,?,?,?,?", 1, 2, 3, 4, 5, function (err, res) {
  if (err) {
    throw err;
  }
  console.log(res);
});
```

Both snippets works well in node.js.

### What is the expected behavior?

```
node test.js 
[ { '$1': 1, '$2': 2, '$3': 3, '$4': 4, '$5': 5 } ]
```

**Comment by Jarred-Sumner** [comment]
Debug logs:
```c++
[SYS] openat(-2, /Users/jarred/Desktop/node_modules/duckdb/lib/binding/) = 5
[fs] close(5)
[napi] napi_open_handle_scope
[napi] napi_open_handle_scope
[napi.cpp:1839] napi_status napi_define_class(napi_env, const char *, size_t, napi_callback, void *, size_t, const napi_property_descriptor *, napi_value *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1150] napi_status napi_create_reference(napi_env, napi_value, uint32_t, napi_ref *)
[napi] napi_close_handle_scope
[napi] napi_open_handle_scope
[napi.cpp:1839] napi_status napi_define_class(napi_env, const char *, size_t, napi_callback, void *, size_t, const napi_property_descriptor *, napi_value *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1150] napi_status napi_create_reference(napi_env, napi_value, uint32_t, napi_ref *)
[napi] napi_close_handle_scope
[napi] napi_open_handle_scope
[napi.cpp:1839] napi_status napi_define_class(napi_env, const char *, size_t, napi_callback, void *, size_t, const napi_property_descriptor *, napi_value *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1150] napi_status napi_create_reference(napi_env, napi_value, uint32_t, napi_ref *)
[napi] napi_close_handle_scope
[napi] napi_open_handle_scope
[napi.cpp:1839] napi_status napi_define_class(napi_env, const char *, size_t, napi_callback, void *, size_t, const napi_property_descriptor *, napi_value *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:1214] napi_status napi_add_finalizer(napi_env, napi_value, void *, napi_finalize, void *, napi_ref *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1150] napi_status napi_create_reference(napi_env, napi_value, uint32_t, napi_ref *)
[napi] napi_close_handle_scope
[napi.cpp:2105] napi_status napi_create_object(napi_env, napi_value *)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:601] napi_status napi_set_named_property(napi_env, napi_value, const char *, napi_value)
[napi] napi_create_string_utf8: IDENTIFIER
[napi] napi_set_element
[napi] napi_create_string_utf8: NUMERIC_CONSTANT
[napi] napi_set_element
[napi] napi_create_string_utf8: STRING_CONSTANT
[napi] napi_set_element
[napi] napi_create_string_utf8: OPERATOR
[napi] napi_set_element
[napi] napi_create_string_utf8: KEYWORD
[napi] napi_set_element
[napi] napi_create_string_utf8: COMMENT
[napi] napi_set_element
[napi.cpp:1150] napi_status napi_create_reference(napi_env, napi_value, uint32_t, napi_ref *)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi.cpp:1978] napi_status napi_create_double(napi_env, double, napi_value *)
[napi] napi_close_handle_scope
[napi.cpp:2317] napi_status napi_set_instance_data(napi_env, void *, napi_finalize, void *)
ASSERTION FAILED: value
/Users/bun/actions-runner/_work/WebKit/WebKit/Source/JavaScriptCore/runtime/JSObjectInlines.h(303) : static bool JSC::JSObject::putInlineForJSObject(JSCell *, JSGlobalObject *, PropertyName, JSValue, PutPropertySlot &)
1   0x102ac43d4 WTFCrash
2   0x10234bc90 JSC::JSValue::toInt32(JSC::JSGlobalObject*) const
3   0x1032c6210 JSC::JSObject::putInlineForJSObject(JSC::JSCell*, JSC::JSGlobalObject*, JSC::PropertyName, JSC::JSValue, JSC::PutPropertySlot&)
4   0x103886d4c llint_slow_path_put_by_id
5   0x102b59910 llint_entry
6   0x114b6c160 5   ???                                 0x0000000114b6c160 0x0 + 4642488672
7   0x102b72fac llint_entry
8   0x102b4ccb4 vmEntryToJavaScript
9   0x1037696b4 JSC::Interpreter::executeCallImpl(JSC::VM&, JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
10  0x1039ddfe4 JSC::call(JSC::JSGlobalObject*, JSC::JSValue, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&, WTF::NakedPtr<JSC::Exception>&)
11  0x1022a33cc Bun::evaluateCommonJSModuleOnce(JSC::VM&, Zig::GlobalObject*, Bun::JSCommonJSModule*, JSC::JSString*, JSC::JSValue, WTF::NakedPtr<JSC::Exception>&)
12  0x1022a8738 Bun::JSCommonJSModule::evaluate(Zig::GlobalObject*, WTF::String const&, ResolvedSource, bool)
13  0x102360b98 Bun::JSCommonJSModule::evaluate(Zig::GlobalObject*, WTF::String const&, ResolvedSource)
14  0x102357300 Bun::fetchCommonJSModule(Zig::GlobalObject*, Bun::JSCommonJSModule*, JSC::JSValue, BunString*, BunString*)
15  0x1022a8300 Bun::jsFunctionRequireCommonJS(JSC::JSGlobalObject*, JSC::CallFrame*)
16  0x114b6003c 15  ???                                 0x0000000114b6003c 0x0 + 4642439228
17  0x102b72fac llint_entry
18  0x114b6c160 17  ???                                 0x0000000114b6c160 0x0 + 4642488672
19  0x102b72fac llint_entry
20  0x102b4ccb4 vmEntryToJavaScript
21  0x1037696b4 JSC::Interpreter::executeCallImpl(JSC::VM&, JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
22  0x1039ddfe4 JSC::call(JSC::JSGlobalObject*, JSC::JSValue, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&, WTF::NakedPtr<JSC::Exception>&)
23  0x1022a33cc Bun::evaluateCommonJSModuleOnce(JSC::VM&, Zig::GlobalObject*, Bun::JSCommonJSModule*, JSC::JSString*, JSC::JSValue, WTF::NakedPtr<JSC::Exception>&)
24  0x1022a8738 Bun::JSCommonJSModule::evaluate(Zig::GlobalObject*, WTF::String const&, ResolvedSource, bool)
25  0x102360b98 Bun::JSCommonJSModule::evaluate(Zig::GlobalObject*, WTF::String const&, ResolvedSource)
26  0x102357300 Bun::fetchCommonJSModule(Zig::GlobalObject*, Bun::JSCommonJSModule*, JSC::JSValue, BunString*, BunString*)
27  0x1022a8300 Bun::jsFunctionRequireCommonJS(JSC::JSGlobalObject*, JSC::CallFrame*)
28  0x114b6003c 27  ???                                 0x0000000114b6003c 0x0 + 4642439228
29  0x102b72fac llint_entry
30  0x114b6c160 29  ???                                 0x0000000114b6c160 0x0 + 4642488672
31  0x102b72fac llint_entry
Stop reason: EXC_BAD_ACCESS (code=1, address=0xbbadbeef)
```

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.0.19
